test/abstractmcmc.jl

```diff
@@ -35,4 +35,23 @@ include("common.jl")
     end
 
     @test m_est ≈ [49 / 24, 7 / 6] atol=RNDATOL
+
+    # Test that using the same AbstractRNG results in the same chain
+    rng1 = MersenneTwister(42)
+    rng2 = MersenneTwister(42)
+    samples1 = AbstractMCMC.sample(
+        rng1, model, κ, metric, adaptor, 10;
+        nadapts = 0,
+        progress=false,
+        verbose=false
+    );
+    samples2 = AbstractMCMC.sample(
+        rng2, model, κ, metric, adaptor, 10;
+        nadapts = 0,
+        progress=false,
+        verbose=false
+    );
+    @test mapreduce(*, samples1, samples2) do s1, s2
+        s1.z.θ == s2.z.θ
+    end # Equivalent to using all, check that all samples are equal
```

Comment:
Why not
```suggestion
    @test all(s1.z.θ .== s2.z.θ)
```
?

Reply:
This would not work no?
And I thought my solution was prettier than 
```
@test all(getindex.(getindex.(samples1, :z), :\theta)) .== getindex.(getindex.(samples2, :z), :\theta))
```